### PR TITLE
Initial state of ECS networking

### PR DESCRIPTION
**Pull request by TheRealChriAnd**

### CODE REVIEW

File: LambdaEngine/Include/Game/ECS/Systems/Networking/ClientSystem.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+	class ClientSystem : public System, protected IClientHandler
+	{
+		friend class EngineLoop;
+
+	public:
+		DECL_UNIQUE_CLASS(ClientSystem);
+		virtual ~ClientSystem();
+
+		bool Connect(IPAddress* address);
```
Comment (Vonsen) [inline LambdaEngine/Include/Game/ECS/Systems/Networking/ClientSystem.h]:
```suggestion
		bool Connect(IPAddress* pAddress);
```

File: LambdaEngine/Source/Game/ECS/Systems/Networking/ClientSystem.cpp [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+	{
+		ClientDesc clientDesc = {};
+		clientDesc.PoolSize = 1024;
+		clientDesc.MaxRetries = 10;
+		clientDesc.ResendRTTMultiplier = 3.0F;
+		clientDesc.Handler = this;
+		clientDesc.Protocol = EProtocol::UDP;
+		clientDesc.PingInterval = Timestamp::Seconds(1);
+		clientDesc.PingTimeout = Timestamp::Seconds(3);
+		clientDesc.UsePingSystem = true;
```
Comment (Vonsen) [inline LambdaEngine/Source/Game/ECS/Systems/Networking/ClientSystem.cpp:42]: spacing

File: LambdaEngine/Include/Networking/API/NetworkSegment.h [hunk 3]
```diff
@@ -37,6 +37,10 @@ namespace LambdaEngine
 			TYPE_ACCEPTED				= UINT16_MAX - 7,
 			TYPE_NETWORK_ACK			= UINT16_MAX - 8,
 			TYPE_NETWORK_DISCOVERY		= UINT16_MAX - 9,
+
+
+			TYPE_ENTITY_CREATE			= UINT16_MAX - 10,
+			TYPE_PLAYER_ACTION			= UINT16_MAX - 11,
```
Comment (Vonsen) [inline LambdaEngine/Include/Networking/API/NetworkSegment.h:43]:
```suggestion
			TYPE_ENTITY_CREATE			= UINT16_MAX - 10,
			TYPE_PLAYER_ACTION			= UINT16_MAX - 11,
```

**Review by Vonsen — CHANGES_REQUESTED** [review]
fix suggested changes, looks good otherwise

File: LambdaEngine/Source/Game/ECS/Systems/Networking/ClientSystem.cpp (hunk 2 above)
Comment (TheRealChriAnd) [reply LambdaEngine/Source/Game/ECS/Systems/Networking/ClientSystem.cpp:42]: Already fixed in another PR

File: LambdaEngine/Include/Networking/API/NetworkSegment.h (hunk 3 above)
Comment (TheRealChriAnd) [reply LambdaEngine/Include/Networking/API/NetworkSegment.h:43]: Spacing should be here since it is temporary Enums

**Review by Vonsen — APPROVED** [review]

**Pull request merged**